File: spring-web-java/src/main/java/spring/web/java/domain/Delivery.java
package spring.web.java.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import spring.web.java.common.constant.DeliveryStatus;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Delivery {

    @Id
    @GeneratedValue
    @Column(name = "delivery_id")
    private Long id;

    @OneToOne(mappedBy = "delivery", fetch = FetchType.LAZY)
    private Order order;

    @Embedded
    private Address address;

    @Enumerated(EnumType.STRING)
    private DeliveryStatus status;

    public static Delivery create(Address address) {
        Delivery delivery = new Delivery();

        delivery.status = DeliveryStatus.READY;
        delivery.address = address;

        return delivery;
    }

    public void setOrder(Order order) {
        this.order = order;
    }

    public void cancel() {
        this.status = DeliveryStatus.CANCEL;
    }
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/Delivery.kt
package spring.web.kotlin.domain

import jakarta.persistence.*
import spring.web.kotlin.constant.DeliveryStatus

@Entity
class Delivery private constructor(
    @Id
    @GeneratedValue
    @Column(name = "delivery_id")
    val id: Long? = null,

    @OneToOne(mappedBy = "delivery", fetch = FetchType.LAZY)
    val order: Order? = null,

    @Embedded
    val address: Address,

    @Enumerated(EnumType.STRING)
    val status: DeliveryStatus
){
    companion object {
        fun create(address: Address) = Delivery(
            status = DeliveryStatus.READY,
            address = address
        )
    }
}